src/pymorize/units.py

```diff
@@ -0,0 +1,108 @@
+import pint
+import re
+
+
+"""
+This module deals with the auto-unit conversion in the cmorization process.
+In case the units in model files differ from CMIP Tables, this module attempts to convert them automatically.
+
+In case of missing units in either model files or CMIP Tables, this module can not convert from a dimentionless base to something with dimension. Dealing with such thing have to done with `action` section in the Rules module on a per variable basis.
+
+"""
+
+
+# TODO: decide a place to hold the missing unit definitions that pint is not aware of
+# from pathlib import Path
+# _unitsfile = Path(__file__).parent / "units_en.txt"
+# u = pint.UnitRegistry(_unitsfile)
+
+
+ureg = pint.UnitRegistry()
+# ureg.define('degC = degree_Celsius')
+# https://ncics.org/portfolio/other-resources/udunits2/
+ureg.define("degrees_east = deg")
+ureg.define("degree_east = deg")
+ureg.define("degrees_north = deg")
+ureg.define("degree_north = deg")
+ureg.define("degrees_west = -1 * deg")
+ureg.define("degrees_south = -1 * deg")
+# https://planetcalc.com/6777/
```

Comment:
I think we should double check here with the model developers. These sorts of factors might be also defined in the model source code, and we should be consistent to avoid changing the simulation results.

For chemistry relevant units specifically it might be good to include a well-known library, e.g. [`chemicals`](https://github.com/CalebBell/chemicals/) as referenced by [Pint's Ecosystem](https://pint.readthedocs.io/en/stable/ecosystem.html).

Reply:
> I think we should double check here with the model developers. These sorts of factors might be also defined in the model source code, and we should be consistent to avoid changing the simulation results.

Thats a valid argument but if "degree_north" is used as a unit, pint does not understand it and so it needs to be defined. "degrees_west" and "degrees_south" is just a counter part in case a conversion is required. As of now, I did not see the need to define the counter parts after going though the tables and though all fesom variable units. I can drop the counter parts now and add them later when the situation demands.

Thanks for pointing out chemicals library. I will look into it.